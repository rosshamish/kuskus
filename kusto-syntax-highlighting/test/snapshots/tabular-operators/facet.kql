// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/facetoperator

StormEvents
| where State startswith "A" and EventType has "Heavy"
| facet by State, EventType
  with 
  (
  where StartTime between(datetime(2007-01-04) .. 7d) 
  | project StartTime, Source, EpisodeId, EventType
  | take 5
  )
