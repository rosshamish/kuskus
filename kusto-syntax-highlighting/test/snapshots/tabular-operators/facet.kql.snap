>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/facetoperator
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>StormEvents
#^^^^^^^^^^^^ source.kusto
>| where State startswith "A" and EventType has "Heavy"
#^^ source.kusto
#  ^^^^^ source.kusto keyword.other.query.kusto
#       ^^^^^^^ source.kusto
#              ^^^^^^^^^^ source.kusto keyword.other.operator.kusto
#                        ^ source.kusto
#                         ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                          ^ source.kusto string.quoted.double.kusto
#                           ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                            ^ source.kusto
#                             ^^^ source.kusto keyword.other.operator.kusto
#                                ^^^^^^^^^^^ source.kusto
#                                           ^^^ source.kusto keyword.other.operator.kusto
#                                              ^ source.kusto
#                                               ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                                ^^^^^ source.kusto string.quoted.double.kusto
#                                                     ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
>| facet by State, EventType
#^^ source.kusto
#  ^^^^^ source.kusto meta.query.facet.kusto keyword.other.query.kusto
#       ^ source.kusto meta.query.facet.kusto
#        ^^ source.kusto meta.query.facet.kusto keyword.other.operator.kusto
#          ^^^^^^^^^^^^^^^^^^ source.kusto
>  with 
#^^ source.kusto
#  ^^^^ source.kusto keyword.other.operator.kusto
#      ^^ source.kusto
>  (
#^^^^ source.kusto
>  where StartTime between(datetime(2007-01-04) .. 7d) 
#^^ source.kusto
#  ^^^^^ source.kusto keyword.other.query.kusto
#       ^^^^^^^^^^^ source.kusto
#                  ^^^^^^^ source.kusto keyword.other.operator.kusto
#                         ^ source.kusto
#                          ^^^^^^^^ source.kusto storage.type.kusto
#                                  ^ source.kusto
#                                   ^^^^ source.kusto constant.numeric.kusto
#                                       ^ source.kusto
#                                        ^^ source.kusto constant.numeric.kusto
#                                          ^ source.kusto
#                                           ^^ source.kusto constant.numeric.kusto
#                                             ^^^^^ source.kusto
#                                                  ^^ source.kusto constant.numeric.kusto
#                                                    ^^^ source.kusto
>  | project StartTime, Source, EpisodeId, EventType
#^^^^ source.kusto
#    ^^^^^^^ source.kusto keyword.other.query.kusto
#           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto
>  | take 5
#^^^^ source.kusto
#    ^^^^ source.kusto meta.query.take.kusto keyword.other.query.kusto
#        ^ source.kusto meta.query.take.kusto
#         ^ source.kusto meta.query.take.kusto constant.numeric.kusto
>  )
#^^^^ source.kusto
>